docs/ga-roadmap.md

```diff
@@ -0,0 +1,172 @@
+* Investigate if any config helper needs to be released as stable, if any do the review of
+the public API, godoc and configuration.
+* Investigate tools that check for API compatibility for go modules, enable them for modules
+that we mark as stable.
+
+### Phase 2
+
+**Tentative Date:** 2021-04-31
+
+**Key Results:** At the end of this phase the Collector’s end-to-end support for OTLP traces
+only be marked as GA.
+
+At the end of this phase we want to ensure that the Collector can be run in production, it can receive
+OTLP trace traffic and emit OTLP trace traffic. The complete list of the packages/modules will be
+finalized during the first part of this phase, but the tentative list is:
+
+* `receiver`
+  * `receiverhelper` - without scraper utils in this phase.
+  * `otlp`
+* `processor`
+  * `processorhelper`
+  * `batch`
+  * `memory_limiter`
+* `exporter`
+  * `exporterhelper`
+  * `otlp`
+  * `otlphttp`
+* `extension`
+  * `extensionhelper`
+  * `healthcheck`
+* `obsreport`
+  * Stabilize the observability metrics (user public metrics).
+
+**Action Items:**
+
+* Create a new milestone for this phase, create issues for all the other action items and add them
+to the milestone.
+* Agreement on all packages/modules that will be marked as stable during this phase.
+* Review all public APIs and godocs for modules that we want to release in this phase.
+  * Fix all critical issues, and remove unnecessary (“When in doubt leave it out”) public APIs.
+  * Remove all already deprecated code from the stable modules.
+* Review all public configuration for all the modules, fix issues.
+* Setup a proper loadtest environment and continuously publish results.
+* Ensure correctness tests produce the expected results, improve until confident that a binary
+that passes them is good to be shipped.
+* Enable security checks on every PR (currently some are ignored like `codeql`).
+
+### Phase 3
+
+Tentative Date: 2021-05-31
+**Key Results:** At the end of this phase all Collector’s core components (receivers,
+processors, exporters, extensions) for traces only will be marked as GA.
+
+At the end of this phase we want to ensure that the Collector can be run in production, it can receive the
+trace traffic and emit OTLP trace traffic. The complete list of the packages/modules will be finalized
+during the first part of this phase, but the tentative list is:
+
+* `receiver`
+  * `jaeger`
+  * `opencensus`
+  * `zipkin`
+* `processor`
+  * `spantransformer` - there are good reasons to merge `attributes` and `span`.
+  * `resource`
+  * `filter` - we will consider offering a filter processor for all telemetry signals not just for metrics
+* `exporter`
+  * `jaeger`
+  * `opencensus`
+  * `zipkin`
+* `extension`
+  * `pprof`
+  * `zpages`
+
+TODO: Add action items list.
+
+### Phase N
+
+TODO: Add more phases if/when necessary.
+
+## Alternatives
+
+One alternative proposal is to try to GA all packages at the same time. This proposal was rejected
```

Comment:
Is the GA here the same as OpenTelemetry GA? Just want to check whether the terms align, for example, is Phase 1 1.0.0, and Phase 2+ GA, meaning the focus is on integration testing, benchmarking, etc?

Reply:
Phase 1 is still 0.x version. Phase 2 will be 1.0.0 GA.